### PR TITLE
Correct tildeB sources and fluxes in ValenciaDivClean

### PR DESCRIPTION
**Pull request by norafl**
## Proposed changes

Corrects the sources and fluxes in magnetic field evolution. 

Currently, the code has made the simplification
$$\partial_i({\tilde B}^j\beta^i) = {\tilde B}^j\partial_i\beta^i,$$
which can introduce errors in handling the divergence of the magnetic field. This update removes the resulting ${\tilde B}^j\partial_i\beta^i$ from the magnetic field source terms and adds a corresponding ${\tilde B}^j\beta^i$ to the corresponding flux terms. This handling is consistent with eq. 20 of https://arxiv.org/pdf/2004.00870 and eq. 32 of https://arxiv.org/pdf/1806.04182.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

### CODE REVIEW

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Fluxes.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ void fluxes_impl(
       tilde_b_flux->get(i, j) =
           tilde_b.get(j) * transport_velocity->get(i) +
           get(lapse) * (get(tilde_phi) * inv_spatial_metric.get(i, j) -
-                        spatial_velocity.get(j) * tilde_b.get(i));
+                        transport_velocity->get(j) * tilde_b.get(i));
```
Comment (wthrowe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Fluxes.cpp]: Should no longer be multiplied by the lapse, I believe.

**Review by wthrowe — CHANGES_REQUESTED** [review]
This still needs updates to the documentation (in `src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp` as well as the `Fluxes.hpp` and `Sources.hpp` files) and the tests.  I haven't checked these specific ones, but most of the tests for this sort of thing have the equations reimplemented in python, which should be the main thing that needs fixing there.

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Fluxes.cpp [hunk 2]
```diff
@@ -66,8 +66,8 @@ void fluxes_impl(
                                 lapse_b_over_w.get(j) * tilde_b.get(i);
       tilde_b_flux->get(i, j) =
           tilde_b.get(j) * transport_velocity->get(i) +
-          get(lapse) * (get(tilde_phi) * inv_spatial_metric.get(i, j) -
-                        spatial_velocity.get(j) * tilde_b.get(i));
+        get(lapse) * (get(tilde_phi) * inv_spatial_metric.get(i, j)) -
+                        transport_velocity->get(j) * tilde_b.get(i);
```
Comment (wthrowe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Fluxes.cpp]: Reorder these terms to match the docs, just for readability.

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp [hunk 3]
```diff
@@ -166,6 +167,9 @@ namespace ValenciaDivClean {
  * divergence-free (no-monopole) condition \f$\Phi = \partial_i {\tilde B}^i =
  * 0\f$.
  *
+ * \note The fluxes and sources of {\tilde B} follow eq. 20 of
```
Comment (wthrowe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp]: Need $s around the math.

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp [hunk 4]
```diff
@@ -166,6 +167,9 @@ namespace ValenciaDivClean {
...
+ * \cite Palenzuela2018 rather than eq. 89 of \cite Moesta2014
```
Comment (wthrowe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp]: The equation numbers for both papers look wrong.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Can you change your wording in commits to be present tense? E.g. `Fix blah` or `Update blah`. It might help to review our contributing guidelines: https://spectre-code.org/contributing_to_spectre.html and the commit message part is noted https://spectre-code.org/contributing_to_spectre.html#git-commit-messages

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Fluxes.cpp [hunk 5]
```diff
@@ -65,9 +65,9 @@ void fluxes_impl(
       tilde_s_flux->get(i, j) = tilde_s.get(j) * transport_velocity->get(i) -
                                 lapse_b_over_w.get(j) * tilde_b.get(i);
       tilde_b_flux->get(i, j) =
-          tilde_b.get(j) * transport_velocity->get(i) +
-          get(lapse) * (get(tilde_phi) * inv_spatial_metric.get(i, j) -
-                        spatial_velocity.get(j) * tilde_b.get(i));
+          tilde_b.get(j) * transport_velocity->get(i)
+        -  transport_velocity->get(j) * tilde_b.get(i) +
+        get(lapse) * (get(tilde_phi) * inv_spatial_metric.get(i, j));
```
Comment (wthrowe) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Fluxes.cpp]: Run this though clang-format.  Their documentation has instructions on how to use it through emacs or git: https://clang.llvm.org/docs/ClangFormat.html .  (Our repo has style options configured in a top-level `.clang-format` file, so you shouldn't need to configure anything yourself.)

**Review by wthrowe — CHANGES_REQUESTED** [review]
This looks good.  Two more things: code formatting, which I've commented on as a line-comment, and squashing.  This PR is pretty much an atomic change, so it should be squashed into a single commit.  Google "git interactive rebase" if you don't know how to do that.  As noted by @nilsdeppe, make sure your resulting commit message follows our guidelines.

I think the CI error is just some random failure, so ignore that if it doesn't happen again after you push the squashed version.

**Comment by wthrowe** [comment]
Formatting looks good now, but something went wrong during your rebase.  Probably easiest to just sort it out in person tomorrow.

**Review by wthrowe — APPROVED** [review]

**Pull request merged**